compile.sh

```diff
@@ -202,7 +202,8 @@ then
   cd ../spdlog || exit
   git checkout v1.9.2
   cd ../googletest || exit
-  git checkout release-1.8.1
+  # older versions have warning as error. commit hash equals HEAD (27.7.2022)
```

Comment:
This comment might look ok now but it does not provide much needed context. Imaging some poor sob (you most likely) has to read that in a few weeks and is trying to figure out what it means.

A suggestion:

"gtest is pinned to this commit because older versions do not build warning free on GCC11 and gtest insists on using -Werror. No release version was available that builds without warnings on GCC11 at this time [2022-07-27]"

On a side note: When writing any date please use ISO-8601 (https://www.iso.org/iso-8601-date-and-time-format.html) Probably the only sane date format ever